aion/web/elm/src/Ranking/View.elm

```diff
@@ -2,56 +2,116 @@ module Ranking.View exposing (..)
 
 import General.Models exposing (Model)
 import Html exposing (..)
+import Html.Attributes exposing(style, src, value)
 import Msgs exposing (Msg(..))
 import Bootstrap.Table as Table
 import Bootstrap.Grid as Grid
+import Bootstrap.Table exposing(rowAttr)
 import Bootstrap.Grid.Col as Col
-import Ranking.Models exposing (PlayerScore, RankingData)
+import Bootstrap.Form as Form
+import Bootstrap.Form.Select as Select
+import Ranking.Models exposing (PlayerScore, RankingData, CategoryRanking)
 import RemoteData exposing (WebData)
+import Urls exposing (host)
+import Navigation exposing (Location)
 
 
 rankingView : Model -> Html Msg
 rankingView model =
     Grid.container []
         [ Grid.row []
-            [ Grid.col [ Col.xs4 ]
-                [ h4 [] [ text "General" ] ]
+            [ Grid.col [ Col.xs4, Col.offsetXs1 ]
+                [ Form.group []
+                    [ Form.label [] [ text "Select Category" ]
+                    , Select.select [ Select.onChange OnRankingCategoryChange ] (selectCategoriesAttributes model)
+                    ]
+                ]
             ]
         , Grid.row []
-            [ Grid.col [ Col.xs4 ]
+            [ Grid.col [ Col.xs10, Col.offsetXs1 ]
                 [ rankingTable model ]
             ]
         ]
 
 
+selectCategoriesAttributes : Model -> List (Select.Item msg)
+selectCategoriesAttributes model =
+    case model.rankingData.data of
+        RemoteData.Success rankingData ->
+            List.map (\category -> Select.item [ value (toString category.categoryId) ] [ text category.categoryName] ) rankingData.rankingList
+        _ -> []
+
+
 rankingTable : Model -> Html Msg
 rankingTable model =
-    Table.simpleTable
-        ( Table.simpleThead
-            [ Table.th [] [ text "User" ]
+    Table.table
+        { options = [ Table.striped, Table.hover ]
+        , thead = Table.simpleThead
+            [ Table.th [] [ text "#" ]
+            , Table.th [ Table.cellAttr(style [ ("width", "5%") ]) ] []
```

Comment:
I would move this to stylesheet. Not sure if this makes sense but the less css we have here the better.

Reply:
I mean, it's just a single parameter. It is much cleaner if we keep it in the same place I think.

Reply:
Both opinions are correct imo, but I think it's much healthier to keep all of the styles in one file, otherwise we may end up not knowing whether some parts are styled in `.css` or in `.elm`. 

Also, suppose you wanted to experiment with layouts, I think it's much easier to change values in `.css` files and observe effects rather than change them in elm files (because in elm files they would be distributed amongst multiple functions.

Another reason is git diff, which will show that you only changed some layout stuff and not necessarily the logic in functions. 

I can agree it may feel like a hustle sometimes, though.